### ISSUE TITLE
Support for raspbian testing on Raspberry Pi?

### ISSUE THREAD

**Issue by cabeca** [50ButtonsEach/fliclib-linux-hci#42]
Hello,

I had flicd up and running fine on a stock raspbian jessie (stable) with 5 paired buttons in my Raspberry Pi.
I recently upgraded raspbian to testing via `apt-get dist-upgrade` (after changing the sources list) and the buttons don't work anymore. If I connect to the flic daemon  with `simpleclient` and issue `getInfo` I get a message saying I have no verified buttons.

```
getInfo
Got info: Attached, aa:aa:aa:aa:aa:aa (PublicBdAddrType), max pending connections: 128, max conns: -1, current pending conns: 0, currently no space: n
No verified buttons yet
```

I verified the sqlite3 file and the buttons table has all the buttons in it.

Can you provide some pointers on how to solve this?

**Comment by Emill** [comment]
The Bluetooth address of the Bluetooth controller must be the same as when
the buttons were paired. Is this the case?

Den 31 dec. 2016 00:33 skrev "Miguel Cabeça" <notifications@github.com>:

**Comment by cabeca** [comment]
I have no idea. aa:aa:aa:aa:aa:aa does not look like a valid address.

**Comment by Emill** [comment]
No that's (for like 99.999% probability) not a valid address.
Can you run "sudo btmon" in a second terminal and then start flicd from the first terminal and tell me what's being printed?

**Comment by Emill** [comment]
Well that was strange...
The Bluetooth controller says it has address AA:AA:AA:AA:AA:AA. Is it a RPi3 you have?
If you pair the buttons now, does it work after a reconnect has been made to the Flics?

**Comment by Emill** [comment]
It seems you're not the only one with this problem: https://www.google.se/search?espv=2&q=%22AA%3AAA%3AAA%3AAA%3AAA%3AAA%22+raspberry+pi+bluetooth

You could also use the "static random address" feature to use another bd addr for the controller. See the readme for details.

**Comment by cabeca** [comment]
So I've figured out the problem with the help of https://www.willhughes.name/post/bluetooth_rpi_xenial/

This blog entry documents a similar problem, but instead of raspbian testing, he used ubuntu xenial.

The Pi 3 needs to load a firmware file for the bluetooth chip to work. All the packages and services that he talks about in his blog post are already in place for raspbian testing, except for the firmware file itself: `BCM43430A1.hcd`

To solve the problem I've put the firmware file in `/lib/firmware/BCM43430A1.hcd`. I also needed to make a symlink from `/lib/firmware` to `/etc/firmware`, because the firmware loader in raspbian testing  expects the firmware to be in `/etc/firmware`.

After a couple of restarts of the service, the buttons work again:

```
getInfo
Got info: Attached, b8:27:eb:70:7d:63 (PublicBdAddrType), max pending connections: 128, max conns: 10, current pending conns: 5, currently no space: n
Verified buttons:
80:e4:da:70:77:93
80:e4:da:70:78:3f
80:e4:da:70:a7:03
80:e4:da:70:bb:03
80:e4:da:70:c4:b1
```

Thanks for your help in solving this issue, completely unrelated to flicd :-)
